repose-aggregator/components/filters/body-extractor-to-header/src/main/scala/org/openrepose/filters/bodyextractortoheader/BodyExtractorToHeaderFilter.scala

```diff
@@ -0,0 +1,152 @@
+  }
+
+  override def doFilter(servletRequest: ServletRequest, servletResponse: ServletResponse, filterChain: FilterChain): Unit = {
+    val mutableHttpRequest = MutableHttpServletRequest.wrap(servletRequest.asInstanceOf[HttpServletRequest])
+
+    def addHeader(name: String, value: String, quality: Option[java.lang.Double], overwrite: Boolean): Unit = {
+      if (overwrite) {
+        mutableHttpRequest.removeHeader(name)
+      }
+      val hdrValue = quality match {
+        case Some(qual) => s"$value;q=$qual"
+        case None => value
+      }
+      mutableHttpRequest.addHeader(name, hdrValue)
+    }
+
+    val jsonDoc: Try[DocumentContext] = {
+      Option(mutableHttpRequest.getContentType) match {
+        case Some(contentType) =>
+          if (contentType.toLowerCase.contains("json")) {
+            val is = mutableHttpRequest.getInputStream
+            if (is.markSupported()) is.mark(0)
+            val jsonString = Source.fromInputStream(is).mkString
+            if (is.markSupported()) is.reset()
+            logger.trace(s"is.available = ${is.available()}")
+            Try(JsonPath.using(jsonPathConfiguration).parse(jsonString))
+          } else {
+            new Failure(new Exception)
+          }
+        case _ =>
+          logger.debug("Content-Type was not defined.")
+          new Failure(new Exception)
+      }
+    }
+
+    extractions.foreach { extraction =>
+      if (jsonDoc.isSuccess) {
+        val extracted = jsonDoc match {
+          case Success(doc) => Try(doc.read[Any](extraction.jsonPath))
+        }
+
+        (extracted, extraction.defaultValue, extraction.nullValue) match {
+          // JSONPath value was extracted AND is NOT Null
+          case (Success(headerValue), _, _) if Option(headerValue).isDefined =>
+            addHeader(extraction.headerName, headerValue.toString, extraction.quality, extraction.overwrite)
+          // JSONPath value was extracted AND is Null AND NullValue is defined
+          case (Success(headerValue), _, Some(nullValue)) =>
+            addHeader(extraction.headerName, nullValue, extraction.quality, extraction.overwrite)
+          // JSONPath value was NOT extracted AND DefaultValue is defined
+          case (Failure(e), Some(defaultValue), _) =>
+            addHeader(extraction.headerName, defaultValue, extraction.quality, extraction.overwrite)
+          case (_, _, _) => // don't add a header
+        }
+      }
+    }
+
+    filterChain.doFilter(mutableHttpRequest, servletResponse)
+  }
+
+  override def destroy(): Unit = {
+    logger.trace("Body Extractor to Header filter destroying...")
+    configurationService.unsubscribeFrom(configurationFile, this.asInstanceOf[UpdateListener[_]])
+    logger.trace("Body Extractor to Header filter destroyed.")
+  }
+
+  override def configurationUpdated(config: BodyExtractorToHeaderConfig): Unit = {
+    extractions = config.getExtraction.asScala.map { extraction =>
```

Comment:
Another style guide thing. I'd drop the period before map. See:
http://docs.scala-lang.org/style/method-invocation.html#higher-order-functions